### PR TITLE
Add backward compatibility for new HIP data types

### PR DESCRIPTION
**Pull request by KKyang**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,6 +32,27 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
...
+  else()
+    execute_process(COMMAND ls -l /etc/alternatives/rocm OUTPUT_VARIABLE ROCmVersion)
+    string(REGEX MATCH "\\/opt/rocm-[0-9]\.[0-9]\.[0-9]\-([0-9]+)" _ ${ROCmVersion})
+  endif()
+  set(ROCmVersion ${CMAKE_MATCH_1})
+endif()
+
+if(ROCmVersion)
+  if(ROCmVersion LESS 13278)
+    add_compile_definitions(HIP_TYPE_BACKWARD_COMPATIBILITY_13278)
```
Comment (jichangjichang) [inline CMakeLists.txt]: Don't bring internal build number version into develop branch

**Comment by jichangjichang** [comment]
Don't merge it, ROCM6.0 already has switch hipDataType. There is no reason to give user chane to use old hip runtime datatype.

File: CMakeLists.txt (hunk 1 above)
Comment (KKyang) [reply CMakeLists.txt]: removed

**Comment by KKyang** [comment]
> Don't merge it, ROCM6.0 already has switch hipDataType. There is no reason to give user chane to use old hip runtime datatype.

It's not old hip runtime data type, it's backward compatibility.

File: CMakeLists.txt [hunk 2]
```diff
@@ -32,6 +32,30 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
...
+    execute_process(COMMAND ls -l /etc/alternatives/rocm OUTPUT_VARIABLE ROCmVersion)
+    string(REGEX MATCH "\\/opt/rocm-([0-9])\.([0-9])\.[0-9]" _ ${ROCmVersion})
+  endif()
+  set(ROCmMajorVersion ${CMAKE_MATCH_1})
+  set(ROCmMinorVersion ${CMAKE_MATCH_2})
+endif()
+
+if(ROCmMajorVersion)
+  if(ROCmMajorVersion EQUAL 6 AND ROCmMinorVersion EQUAL 0)
+    add_compile_definitions(HIP_TYPE_BACKWARD_COMPATIBILITY)
```
Comment (jichangjichang) [inline CMakeLists.txt:53]: Formal release rocm6.0 has hipDataType HIP_R_E4M3_FNUZ.

**Pull request closed without merging**